### DISCUSSION TITLE
Question about tokenizer_manager.py line 200 : Regarding the preprocessing of multimodal models

### DISCUSSION THREAD

**Discussion by CSEEduanyu** [sgl-project/sglang#1824] [Q&A]
https://github.com/sgl-project/sglang/blob/6fcd6d7d6dec7aea858d7441effd8a04b6d05474/python/sglang/srt/managers/tokenizer_manager.py#L200

**Comment by CSEEduanyu** [comment]
Question 1: According to the code here, the multi-modal inputs_ids is generated in the preprocessing stage, and the process is asynchronous. So the infer of the visual model and the infer of the LLM may happen at the same time. Will there be any conflict?

**Comment by CSEEduanyu** [comment]
Question 2: In the few multimodal models that have been implemented so far, process_images_async() does not return the input_ids field, so I am confused by the code on line 200 here.

**Comment by CSEEduanyu** [comment]
My current dilemma is that I want to add support for a new multimodal model, but I don't know whether process_images_async() should return the input_ids of the image mappings or the raw pixel information?

**Comment by merrymercy** [comment]
You can learn from existing multimodal examples (llama 3.2 and llava).
https://github.com/sgl-project/sglang/blob/f407fcf9ef0ef637deb6b62cd9044e1778c53b89/python/sglang/srt/managers/tokenizer_manager.py#L206-L210
